junto-rust/zomes/core/code/src/expressions/indexing.rs

```diff
@@ -0,0 +1,266 @@
+//Module to handle all channel related operations
+use hdk::{
+    error::ZomeApiResult,
+    error::ZomeApiError,
+    holochain_core_types::{
+        entry::Entry, 
+        cas::content::Address,
+        hash::HashString
+    }
+};
+
+use std::collections::HashMap;
+use itertools::Itertools;
+use std::convert::TryFrom;
+
+use super::definitions::{
+    app_definitions,
+    function_definitions::{
+        FunctionDescriptor,
+        FunctionParameters
+    }
+};
+use super::group;
+use super::user;
+use super::channel;
+
+pub fn create_query_points(query_points: Vec<HashMap<String, String>>, context: &Address, privacy: &app_definitions::Privacy, 
+                            query_type: &String, expression: &Address) -> ZomeApiResult<String>{
+    let mut addressed_params: Vec<HashMap<String, String>> = query_points.to_vec();
+    //hdk::api::link_entries(&context, &expression, "expression")?;
+    if context != &HashString::from(hdk::api::DNA_ADDRESS.to_string()){
+        let context_entry = hdk::get_entry(&context)?;
+        match context_entry {
```

Comment:
You can use `utils::get_as_type` instead of the double nested match expression to check if it is an app entry and is a particular type